docs/diagrams/FilterPredicatesClassDiagram.puml

```diff
@@ -0,0 +1,77 @@
+@startuml
```

Comment:
imo can omit the method names since most of them repeat, you could use the below chunk of code to style the diagram to be more concise, similar to how the default class diagrams look.

```
!include style.puml
skinparam arrowThickness 1.1
skinparam arrowColor MODEL_COLOR
skinparam classBackgroundColor MODEL_COLOR 
```